src/SFA.DAS.LevyTransferMatching.Web/Validators/Location/LocationValidatorService.cs

```diff
@@ -15,31 +16,52 @@ public LocationValidatorService(ILocationService locationService)
             _locationService = locationService;
         }
 
-        public async Task<Dictionary<int, string>> ValidateLocations(LocationPostRequest request)
+        public async Task<Dictionary<int, string>> ValidateLocations(LocationPostRequest request, IDictionary<int, IEnumerable<string>> multipleValidResults)
         {
             var errors = new Dictionary<int, string>();
 
-            await CheckLocationsExist(errors, request.Locations);
+            await CheckLocationsExist(errors, request.Locations, multipleValidResults);
             CheckForDuplicates(errors, request.Locations);
 
             return errors;
         }
 
-        private async Task CheckLocationsExist(Dictionary<int, string> errors, List<string> locations)
+        private async Task CheckLocationsExist(Dictionary<int, string> errors, List<string> locations, IDictionary<int, IEnumerable<string>> multipleValidResults)
```

Comment:
This method is too complex - could you see if it could be refactored? It's really difficult to follow. The intent should be clear enough that comments aren't required.

See also: SonarCloud assessment: `Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed.`

Reply:
Yeah, it was a bit of a mess to be fair. I've refactored this.